### PR TITLE
Added support for creating token (bank account, credit card).

### PR DESCRIPTION
**Pull request by sudiptpa**
Hey @rotassator 

I need this real quick for my production, i have pushed all tested code to this repository after testing with my project, if you could review, merge, tag new release that would be really great.

Thanks heaps for this package !

### CODE REVIEW

File: src/Message/CreateSingleUseCardTokenRequest.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * @link https://www.payway.com.au/docs/rest.html#tokenise-a-credit-card
+ */
+class CreateSingleUseCardTokenRequest extends AbstractRequest
+{
+    public function getData()
+    {
+        $this->getCard()->validate();
+
+        // PayWay requires two digit expiry month.
+        $expiryDateMonth = $this->getCard()->getExpiryMonth() > 9
```
Comment (joshbenham) [inline src/Message/CreateSingleUseCardTokenRequest.php]: You can use str_pad to make it a little cleaner.

```php
$expiryDateMonth = str_pad($this->getCard()->getExpiryMonth(), 2, 0, STR_PAD_LEFT);
```

Comment (sudiptpa) [reply src/Message/CreateSingleUseCardTokenRequest.php]: Thanks for the review @joshbenham i will have a look and update it.

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2016 Steve Dixon
```
Comment (rotassator) [inline LICENSE]: The year on a copyright statement indicates when the license was established, not just the current year. I think we'll leave it at 2016.

File: src/Gateway.php [hunk 3]
```diff
@@ -18,7 +18,7 @@ public function getDefaultParameters()
         return array(
             'apiKeyPublic' => '',
             'apiKeySecret' => '',
-            'merchantId'   => '',
+            'merchantId' => '',
```
Comment (rotassator) [inline src/Gateway.php]: I had an established style where separators are lined up on successive lines. It's a little thing, but I don't see the point of changing things without reason. Was this changed on purpose?

**Review by rotassator — CHANGES_REQUESTED** [review]
Hi Sujip. Sorry, have been AFK for a bit. 

The changes look OK to me, but I don't have a handy way of testing them at the moment. Can you confirm that everything works OK?

I've added a couple of minor review comments that would be great to quickly fix up, but I'm happy to approve when you confirm that it's all working. 

PS. I didn't implement this kind of tokenisation before as it means that you have to deal with handling credit card / bank account details, and are therefore subject to PCI-DSS Compliance rules and laws. Have you considered those in your project?

**Comment by sudiptpa** [comment]
@rotassator Thanks for responding 

Regarding the PCI-DSS Compliance rules and laws, there is a message on their documentation saying :  (Red Message Box at top) Your PCI-DSS compliance scope is reduced when your server does not store or process credit card details. (https://www.payway.com.au/docs/rest.html#single-use-tokens). So we don't store the bank, credit card stuff, but simply send them to PayWay servers to handle them and respond us back.

And, about the testing we have tested them fully and are production ready, both credit card token, bank token working really fine. I've added the example note in the README to show how to use the those features.

File: src/Gateway.php (hunk 3 above)
Comment (sudiptpa) [reply src/Gateway.php]: Ok, i will fix that back, its the editor that fixed it, as my config for the editor was like that to fix automatically.

File: LICENSE (hunk 2 above)
Comment (sudiptpa) [reply LICENSE]: Sure, i got you, will put it back to the original one.

**Review by rotassator — APPROVED** [review]

**Comment by rotassator** [comment]
@sudiptpa I've created a new release 1.8.0 that includes your changes. Thanks! 
https://github.com/rotassator/omnipay-payway-restapi/releases/tag/1.8.0

**Comment by sudiptpa** [comment]
Thanks a lot for the new tag release, i will try to update the code to v3 support for Omnipay later when i get time.

**Pull request merged**